pkg/pushsync/pushsync.go

```diff
@@ -153,141 +154,99 @@ func (ps *PushSync) handler(ctx context.Context, p p2p.Peer, stream p2p.Stream)
 			_ = stream.FullClose()
 		}
 	}()
+
 	var ch pb.Delivery
 	if err = r.ReadMsgWithContext(ctx, &ch); err != nil {
 		return fmt.Errorf("pushsync read delivery: %w", err)
 	}
+
 	ps.metrics.TotalReceived.Inc()
 
 	chunk := swarm.NewChunk(swarm.NewAddress(ch.Address), ch.Data)
 	chunkAddress := chunk.Address()
 
-	span, logger, ctx := ps.tracer.StartSpanFromContext(ctx, "pushsync-handler", ps.logger, opentracing.Tag{Key: "address", Value: chunkAddress.String()})
+	span, _, ctx := ps.tracer.StartSpanFromContext(ctx, "pushsync-handler", ps.logger, opentracing.Tag{Key: "address", Value: chunkAddress.String()})
 	defer span.Finish()
 
 	stamp := new(postage.Stamp)
-	// attaching the stamp is required becase pushToClosest expects a chunk with a stamp
 	err = stamp.UnmarshalBinary(ch.Stamp)
 	if err != nil {
 		return fmt.Errorf("pushsync stamp unmarshall: %w", err)
 	}
 	chunk.WithStamp(stamp)
 
 	if cac.Valid(chunk) {
-		if ps.unwrap != nil {
-			go ps.unwrap(chunk)
-		}
+		go ps.unwrap(chunk)
 	} else if !soc.Valid(chunk) {
 		return swarm.ErrInvalidChunk
 	}
 
 	price := ps.pricer.Price(chunkAddress)
 
-	// if the peer is closer to the chunk, AND it's a full node, we were selected for replication. Return early.
-	if p.FullNode {
-		if closer, _ := p.Address.Closer(chunkAddress, ps.address); closer {
+	store := func(ctx context.Context) error {
```

Comment:
could this method be moved to individual function, out of `handler`? `handler` looks quite big atm, so every reduction in it's size would simplify reading.